src/include/reconcile.h

```diff
@@ -297,6 +297,14 @@ struct __wt_reconcile {
     bool rec_page_cell_with_ts;
     bool rec_page_cell_with_txn_id;
     bool rec_page_cell_with_prepared_txn;
+
+    /*
+     * When removing a key due to a tombstone with a durable timestamp of "none", we also remove the
+     * history store contents associated with that key. Keep the pertinent state here: a flag to say
+     * whether this is appropriate, and a cached history store cursor for doing it.
+     */
+    bool hs_clear_on_tombstone;
+    WT_CURSOR *hs_cursor;
```

Comment:
It has longer lifetime than the call to the new function. The reconcile structure isn't a great place, granted, but it does have the correct lifespan...

I don't think we don't want to go back to pasting the logic for the hs-clear flag three times. It could be moved entirely inside the new function, but at the cost of recomputing it every time (probably not important) and not being able to use it to skip the call (less clear).

The history cursor is supposed to persist from one row to the next once created (that's how the pasted-three-times logic worked) -- it could be moved inside the new function at the cost of creating and destroying it for every affected row. That seems undesirable, but maybe it's not that expensive?

Reply:
I can live with it.